src/resources.rs

```diff
@@ -45,14 +47,40 @@ enum Repr {
   Stdout(tokio::io::Stdout),
   Stderr(tokio::io::Stderr),
   FsFile(tokio::fs::File),
+  TcpListener(tokio::net::TcpListener),
+  TcpStream(tokio::net::TcpStream),
 }
 
 // Abstract async file interface.
 // Ideally in unix, if Resource represents an OS rid, it will be the same.
+#[derive(Debug)]
 pub struct Resource {
   pub rid: ResourceId,
 }
 
+impl Resource {
+  // TODO Should it return a Resource instead of net::TcpStream?
```

Comment:
IDK, I can't see through it at this point. Let's leave it for now and refactor (if necessary) when we add other types of servers.